bin/bulk.dart

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Example uses:
+///   bulk analyze --output output-dir-1 pkg1 pkg2 pkg3
+///   bulk analyze --dart-sdk /path/to/sdk --output output-dir-2 pkg1 pkg2 pkg3
+///   bulk summary output-dir-1 output-dir-2
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:pana/pana.dart';
+import 'package:pool/pool.dart';
+
+final _jsonEncoder = new JsonEncoder.withIndent('  ');
+
+main(List<String> args) async {
+  final runner = new CommandRunner('bulk', 'Pana bulk processing')
+    ..addCommand(new AnalyzeCommand())
+    ..addCommand(new SummaryCommand());
+  await runner.run(args);
+}
+
+class AnalyzeCommand extends Command {
+  @override
+  String get name => 'analyze';
+
+  @override
+  String get description => 'Analyzes multiple packages';
+
+  AnalyzeCommand() {
+    argParser
+      ..addOption('concurrency',
+          defaultsTo: '1', help: 'The concurrent execution of the analyis.')
+      ..addOption('dart-sdk', help: 'The directory of the Dart SDK.')
+      ..addOption('flutter-sdk', help: 'The directory of the Flutter SDK.')
+      ..addOption('input', help: 'The input file with the list of packages.')
+      ..addOption('output',
+          help: 'The output directory to store the per-package results.');
+  }
+
+  @override
+  Future run() async {
+    final concurrency = int.tryParse(argResults['concurrency'] as String) ?? 1;
+    final pool = new Pool(concurrency);
+
+    final packages = new Set<String>();
+    packages.addAll(argResults.rest);
+
+    final String inputFileName = argResults['input'];
+    if (inputFileName != null) {
+      final inputFile = new File(inputFileName);
+      if (!(await inputFile.exists())) {
+        print('Input file $inputFileName does not exists.');
+        exit(-1);
+      }
+      final lines = await inputFile.readAsLines();
+      packages.addAll(lines.where((l) => l.isNotEmpty));
+    }
+
+    if (packages.isEmpty) {
+      print('No package to analyze.');
+      exit(-1);
```

Comment:
exit(1). Or possibly exit(0), this is not bad condition. Could we proceed through the below code as a no-operation and provide the normal output with information about zero packages?